be/src/olap/rowset/segment_v2/inverted_index_writer.cpp

```diff
@@ -0,0 +1,421 @@
+            }
+            return Status::InternalError("field type not supported");
+        } catch (const CLuceneError& e) {
+            LOG(WARNING) << "Inverted index writer init error occurred: " << e.what();
+            return Status::Error<doris::ErrorCode::INVERTED_INDEX_NOT_SUPPORTED>();
+        }
+    }
+
+    void close() {
+        if (_index_writer) {
+            _index_writer->close();
+            _CLLDELETE(_index_writer)
+            _index_writer = nullptr;
+        }
+
+        if (_doc) {
+            _CLLDELETE(_doc)
+            _doc = nullptr;
+        }
+
+        if (_analyzer) {
+            _CLLDELETE(_analyzer)
+            _analyzer = nullptr;
+        }
+
+        if (_char_string_reader) {
+            _CLDELETE(_char_string_reader)
+            _char_string_reader = nullptr;
+        }
+    };
+
+    Status init_fulltext_index() {
+        bool create = true;
+
+        auto index_path = InvertedIndexDescriptor::get_temporary_index_path(
+                _directory + "/" + _segment_file_name, _index_meta->index_id());
+
+        // LOG(INFO) << "inverted index path: " << index_path;
+
+        if (lucene::index::IndexReader::indexExists(index_path.c_str())) {
+            create = false;
+            if (lucene::index::IndexReader::isLocked(index_path.c_str())) {
+                LOG(INFO) << ("Lucene Index was locked... unlocking it.\n");
+                lucene::index::IndexReader::unlock(index_path.c_str());
+            }
+        }
+
+        _char_string_reader = _CLNEW lucene::util::SStringReader<char>;
+        _doc = _CLNEW lucene::document::Document();
+        _dir.reset(DorisCompoundDirectory::getDirectory(_fs, index_path.c_str(), true));
+
+        if (_parser_type == InvertedIndexParserType::PARSER_STANDARD) {
+            _analyzer = _CLNEW lucene::analysis::standard::StandardAnalyzer();
+            _index_writer = _CLNEW lucene::index::IndexWriter(_dir.get(), _analyzer, create, true);
+        } else if (_parser_type == InvertedIndexParserType::PARSER_ENGLISH) {
+            _analyzer = _CLNEW lucene::analysis::SimpleAnalyzer<char>();
+            _index_writer = _CLNEW lucene::index::IndexWriter(_dir.get(), _analyzer, create, true);
+        } else if (_parser_type == InvertedIndexParserType::PARSER_CHINESE) {
+            auto chinese_analyzer = _CLNEW lucene::analysis::LanguageBasedAnalyzer();
+            chinese_analyzer->setLanguage(L"chinese");
+            chinese_analyzer->initDict(config::dict_path);
+            _analyzer = chinese_analyzer;
+            _index_writer = _CLNEW lucene::index::IndexWriter(_dir.get(), _analyzer, create, true);
+        } else {
+            // ANALYSER_NOT_SET, ANALYSER_NONE use default SimpleAnalyzer
+            _analyzer = _CLNEW lucene::analysis::SimpleAnalyzer<TCHAR>();
+            _index_writer = _CLNEW lucene::index::IndexWriter(_dir.get(), _analyzer, create, true);
```

Comment:
duplicate code for new _index_writer can be moved to one place after all branches.